content/blog/prototyping/index.md

```diff
@@ -0,0 +1,755 @@
+
+## Summary
+
+Let's see how Rust stacks up against Python for prototyping:
+
+| Aspect                    | Python                                                                                                                             | Rust                                                                                                                      |
+| ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| Initial Development Speed | ✓ Very quick to write initial code<br>✓ No compilation step<br>✓ Simple syntax                                                     | ⚠️ Slightly slower initial development<br>✓ Type inference helps<br>✓ Tools like `bacon` provide quick feedback            |
+| Error Handling            | ❌ Silent failures possible<br>❌ Runtime exceptions<br>❌ Type errors discovered late                                                | ✓ Explicit error handling (with `unwrap` for prototypes)<br>✓ Compile-time error catching<br>✓ Type safety from the start |
+| Transition to Production  | ❌ Need extensive testing to catch type errors<br>❌ Often requires rewrite in another language<br>❌ Performance optimization needed | ✓ Same code can go to production<br>✓ Already has good performance<br>✓ Memory safety guaranteed                          |
+| Maintenance               | ❌ Type errors surface during runtime<br>❌ Refactoring is risky<br>❌ Documentation can get out of sync                              | ✓ Compiler catches most issues<br>✓ Safe refactoring with type system<br>✓ Types serve as documentation                   |
+| Debugging                 | ✓ Interactive debugger<br>✓ Easy print debugging                                                            | ✓ `dbg!` macro<br>✓ Compile-time catches most bugs                                                                                                 |
+| Code Evolution            | ❌ Hard to maintain large codebases<br>❌ Type issues compound                                                                      | ✓ Compiler guides improvements<br>✓ Types help manage complexity                         |
+| Collaboration             | ❌ Implicit assumptions about types<br>❌ Easy to break others' code<br>✓ Lower learning curve                                       | ✓ Explicit contracts through types<br>✓ Compiler prevents many conflicts<br>⚠️ Steeper learning curve                      |
+
+The beauty of prototyping in Rust is that your "rough drafts" are often already production-worthy.
+Even when I liberally use `unwrap()`, stick everything in `main.rs`, and reach for owned types everywhere, the resulting code is still memory-safe and reasonably fast.
```

Comment:
"production-worthy" is absolutely the wrong phrase here. Production ought to be rock solid, and my `.unwrap()` laden prototype isn't. But it is on-par with my Python prototype in reliability, but outperforms it easily, making the experiment in production potentially cheaper.